resources/assets/js/annotations/components/annotationCanvas.vue

```diff
@@ -679,6 +676,26 @@ export default {
                 Keyboard.on('Backspace', this.deleteLastCreatedAnnotation, 0, this.listenerSet);
             }
         },
+        isInvalidShape(feature) {
+            let geometry = feature.getGeometry();
+            let points = [];
+            switch (geometry.getType()) {
+                case 'Circle':
+                    return parseInt(geometry.getRadius()) === 0;
+                case 'LineString':
+                    points = geometry.getCoordinates();
+                    return (new Set(points.map(xy => String([xy])))).size < 2;
+                case 'Rectangle':
+                case 'Ellipse':
+                    points = geometry.getCoordinates()[0];
+                    return (new Set(points.map(xy => String([xy])))).size !== 4;
+                case 'Polygon':
+                    points = geometry.getCoordinates()[0];
+                    return (new Set(points.map(xy => String([xy])))).size < 3;
+                default:
+                    return false;
+            }
+        },
```

Comment:
You can reuse the function from `videos/utils.js`. Usually the video annotation tool reuses stuff from the image annotation tool, so maybe you can put the function into an `annotations/utils.js` and import it from there?